resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java

```diff
@@ -1354,4 +1369,94 @@ public String checkAidStatus(String aid) throws ResidentServiceCheckedException
+			Map<String, String> eventStatusMap;
+			if (requestType != null){
+				eventStatusMap = requestType.getAckTemplateVariables(templateUtil, eventId);
+
+				EventStatusResponseDTO eventStatusResponseDTO = new EventStatusResponseDTO();
+				eventStatusResponseDTO.setEventId(eventId);
+				eventStatusResponseDTO.setEventType(requestType.name());
+				eventStatusResponseDTO.setEventStatus(getEventStatusForRequestType(residentTransactionEntity.get().getStatusCode()));
+				eventStatusResponseDTO.setIndividualId(getIndividualIdType());
+				eventStatusResponseDTO.setSummary(residentTransactionEntity.get().getRequestSummary());
+				eventStatusResponseDTO.setTimestamp(residentTransactionEntity.get().getCrDtimes().toString());
+
+				eventStatusMap.remove("eventId");
+				eventStatusMap.remove("eventType");
+				eventStatusMap.remove("eventStatus");
+				eventStatusMap.remove("individualId");
+				eventStatusMap.remove("summary");
+				eventStatusMap.remove("timestamp");
+
+				for (Map.Entry<String, String> entry : eventStatusMap.entrySet()) {
+					entry.setValue(getResidentTransactionEntityValue(residentTransactionEntity, entry.getKey()));
+				}
+				eventStatusResponseDTO.setInfo(eventStatusMap);
+				responseWrapper.setId(serviceEventId);
+				responseWrapper.setVersion(serviceEventId);
+				responseWrapper.setResponsetime(LocalDateTime.parse(LocalDateTime.now().toString()));
+				responseWrapper.setResponse(eventStatusResponseDTO);
+			}
+		}
+		catch (Exception e){
+			logger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.APPLICATIONID.toString(),
+					LoggerFileConstant.APPLICATIONID.toString(),
+					"ResidentServiceImpl::getEventStatus():: Exception");
+			throw new ResidentServiceCheckedException(ResidentErrorCode.EVENT_STATUS_NOT_FOUND);
+		}
+		return responseWrapper;
+	}
+
+	private String getIndividualIdType() throws ApisResourceAccessException {
+		String individualId= identityServiceImpl.getResidentIndvidualId();
+		if(requestValidator.validateUin(individualId)){
+			return UIN;
+		} else if(requestValidator.validateVid(individualId)){
+			return VID;
+		} else{
+			return AID;
+		}
+	}
+
+	private String getEventStatusForRequestType(String statusCode) {
+		String eventStatus = "";
+		List<String> successStatusCodes = Arrays.stream(EventStatusSuccess.values()).map(Enum::name).collect(Collectors.toList());
+		List<String> failureStatusCodes = Arrays.stream(EventStatusFailure.values()).map(Enum::name).collect(Collectors.toList());
+		if(successStatusCodes.contains(statusCode)){
+			eventStatus = EventStatus.SUCCESS.name();
+		} else if(failureStatusCodes.contains(statusCode)){
+			eventStatus = EventStatus.FAILED.name();
+		} else {
+			eventStatus = EventStatus.IN_PROGRESS.name();
+		}
+		return eventStatus;
+	}
+
+	private String getResidentTransactionEntityValue(Optional<ResidentTransactionEntity> residentTransactionEntity, String key) {
+		String value = null;
+		switch (key) {
+
+		case "purpose":
```

Comment:
Please create constats for these strings

Reply:
yes done.